pyanaconda/modules/subscription/subscription_interface.py

```diff
@@ -0,0 +1,152 @@
+#
+# DBus interface for the subscription module.
+#
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.modules.common.constants.services import SUBSCRIPTION
+from pyanaconda.dbus.property import emits_properties_changed
+from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
+from pyanaconda.modules.common.base import KickstartModuleInterface
+from pyanaconda.dbus.interface import dbus_interface
+
+
+@dbus_interface(SUBSCRIPTION.interface_name)
+class SubscriptionInterface(KickstartModuleInterface):
+    """DBus interface for Subscription module."""
+
+    def connect_signals(self):
+        super().connect_signals()
+        self.watch_property("Role", self.implementation.role_changed)
+        self.watch_property("SLA", self.implementation.sla_changed)
+        self.watch_property("Usage", self.implementation.usage_changed)
+        self.watch_property("Addons", self.implementation.addons_changed)
+
+        # the SystemPurposeWillBeSet property depends on the value of
+        # all other system purpose properties
+        self.watch_property("IsSystemPurposeSet", self.implementation.role_changed)
+        self.watch_property("IsSystemPurposeSet", self.implementation.sla_changed)
+        self.watch_property("IsSystemPurposeSet", self.implementation.usage_changed)
+        self.watch_property("IsSystemPurposeSet", self.implementation.addons_changed)
```

Comment:
This code reveals the implementation of the property `is_system_purpose_set`. Please, create a new signal `is_system_purpose_set_changed` in the module and handle the connections to other signals there.

Reply:
Indeed, that looks cleaner!